tests/components/test_remote_homeassistant.py

```diff
@@ -0,0 +1,84 @@
+"""The tests for the remote_homeassistant platform."""
+import asyncio
+from unittest.mock import patch
+
+from aiohttp import WSMsgType
+from async_timeout import timeout
+import pytest
+
+from homeassistant.components import remote_homeassistant
+from homeassistant.core import callback
+from homeassistant.components import websocket_api as wapi, frontend
+from homeassistant.setup import async_setup_component, setup_component
+from homeassistant.components import http
+
+from tests.common import (
+    get_test_home_assistant,
+    mock_mqtt_component,
+    fire_mqtt_message,
+    mock_state_change_event,
+    fire_time_changed
+)
+
+instance01_port = 8124
+instance02_port = 8125
+
+class TestRemoteHomeassistant(object):
```

Comment:
I suggest you go with pytest tests that are not class based. That will be cleaner and you probably want to use the async home assistant api, so then write tests that are coroutines.